pkg/insights/insightsclient/insightsclient.go

```diff
@@ -114,6 +115,16 @@ func clientTransport(authorizer Authorizer) http.RoundTripper {
 	return transport.DebugWrappers(clientTransport)
 }
 
+func UserAgent(releaseVersionEnv string, v apimachineryversion.Info, cv *configv1.ClusterVersion) string {
+	gitVersion := v.GitVersion
+	// If the RELEASE_VERSION is set in pod, use it
+	if releaseVersionEnv != "" {
+		gitVersion = releaseVersionEnv
+	}
+	gitVersion = fmt.Sprintf("%s-%s", gitVersion, v.GitCommit)
+	return fmt.Sprintf("insights-operator/%s cluster/%s", gitVersion, cv.Spec.ClusterID)
```

Comment:
Have you tried with real cluster, when trying against one, I'm getting `v0.0.0-master+$Format:%h$-$Format:%H$` in the `gitVersion`, which doesn't look right.

Reply:
How can I try it on a 'real cluster'. I always run it locally while pointing at a cluster created by cluster-bot

Reply:
So after a bit of investigation the build command doesn't seem to work as intended. The values the `version` returns are the defaults for that package and not the special ones that we should be setting during build.

`build:
	go build -mod=vendor -ldflags "-X vendor/k8s.io/client-go/pkg/version.gitCommit=$$(git rev-parse HEAD) -X vendor/k8s.io/client-go/pkg/version.gitVersion=v1.0.0+$$(git rev-parse --short=7 HEAD)" -o bin/insights-operator ./cmd/insights-operator`

Reply:
Yes, that is right. That is why this change prefers RELEASE_VERSION envvar. If the var is set, it overrides the value built-in the package. Setting it should then affect Debug as well.
At some point later we would also like to resolve filling version in the ldflags, but I would do it in a separate CL.

Reply:
I tested by using the envvar, which works but it still appends the `v.GitCommit` to the the value of the envvar and that will be`"$Format:%H$"`

Reply:
When you build it with make build, it should set the gitVersion in the binary, when you run the binnary afterwards, will it have it set right ? It should.

Reply:
conclusion: the `make build` had incorrect path to the packages where it tried to set the variables.

incorrect path: `github.com/openshift/insights-operator/vendor/k8s.io/client-go/pkg/version.gitCommit`
correct path: `k8s.io/client-go/pkg/version.gitCommit`